crates/compiler/fmt/src/test_helpers.rs

```diff
@@ -0,0 +1,80 @@
+use bumpalo::Bump;
+use roc_test_utils::assert_multiline_str_eq;
+
+use crate::{
+    annotation::{Formattable, Newlines, Parens},
+    Buf,
+};
+
+/// Check that the given input is formatted to the given output.
+/// If the expected output is `None`, then this only checks that the input
+/// parses without error, formats without error, and
+/// (optionally, based on the value of `check_stability`) re-parses to
+/// the same AST as the original.
```

Comment:
nit: iirc this property is usually known "idempotency". no strong opinion either way, not sure which one is actually more familiar

Reply:
Good point, thanks!